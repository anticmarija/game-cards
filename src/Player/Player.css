.player {
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  position: absolute;
  bottom: 15px;
  width: 100%;
}

.player-score {
  position: absolute;
  top: -60px;
  color: yellowgreen;
  font-size: 24px;
}

.card-image {
  width: 80px;
  height: 120px;
  margin-right: 2px;
}

@media only screen and (max-width: 930px) {
  .card-image {
    width: 75px;
    height: 110px;
    margin-right: -25px;
  }
}

@media only screen and (max-width: 530px) {
  .card-image {
    width: 55px;
    height: 80px;
    margin-right: -15x;
  }

  .player-score {
    top: -40px;
    font-size: 16px;
  }
}
