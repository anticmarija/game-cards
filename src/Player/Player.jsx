import React from 'react';
import { connect } from 'react-redux';
import './Player.css';
import { throwCards } from '../store/actions/game.actions';
import {
  getMoveInProgress,
  getNumberOfPlayers,
  getUserPlayerCards,
} from '../store/selectors/selectors';

export const Player = ({
  player,
  throwCards,
  moveInProgress,
  numberOfPlayers,
}) => {
  return (
    <div>
      {player &&
        player.cards.map(card => (
          <img
            data-testid={card.code}
            onClick={() => {
              if (!moveInProgress) {
                throwCards({ user: 'player1', card: card }, numberOfPlayers);
              }
            }}
            className="card-image"
            key={card.code}
            src={card.image}
            alt={card.code}
          />
        ))}
    </div>
  );
};

export const mapStateToProps = state => ({
  player: getUserPlayerCards(state),
  moveInProgress: getMoveInProgress(state),
  numberOfPlayers: getNumberOfPlayers(state),
});

export const mapDispatchToProps = dispatch => ({
  throwCards: (cardThrownData, numberOfPlayers) =>
    dispatch(throwCards(cardThrownData, numberOfPlayers)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Player);
