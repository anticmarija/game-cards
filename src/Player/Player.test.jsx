import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Player, mapDispatchToProps } from './Player';

const props = {
  player: {
    cards: [
      { image: 'image', code: 'code' },
      { image: 'image2', code: 'code2' },
    ],
  },
  throwCards: jest.fn(),
  moveInProgress: false,
  numberOfPlayers: 2,
};

describe('Test Player component', () => {
  it('should render correctly', () => {
    const { container } = render(<Player {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should throw card on click', () => {
    const { queryByTestId } = render(<Player {...props} />);

    const card = queryByTestId('code');

    fireEvent.click(card);

    expect(props.throwCards).toHaveBeenCalled();
  });

  it('should not throw card on click if move is in progress', () => {
    const changedProps = {
      ...props,
      throwCards: jest.fn(),
      moveInProgress: true,
    };
    const { queryByTestId } = render(<Player {...changedProps} />);

    const card = queryByTestId('code');

    fireEvent.click(card);

    expect(changedProps.throwCards).toHaveBeenCalledTimes(0);
  });
});

describe('mapDispatchToProps', () => {
  const dispatchMock = jest.fn();

  it('should provide a prop to throwCards', () => {
    const dispatchProps = mapDispatchToProps(dispatchMock);

    dispatchProps.throwCards({ user: 'player1', card: { value: 'ACE' } }, 2);
    expect(dispatchMock).toHaveBeenCalled();
  });
});
