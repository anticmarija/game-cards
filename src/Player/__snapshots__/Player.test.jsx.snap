// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Player component should render correctly 1`] = `
<div
  class="player"
>
  <span
    class="player-score"
  >
    Player 1: 
  </span>
  <img
    alt="code"
    class="card-image"
    data-testid="code"
    src="image"
  />
  <img
    alt="code2"
    class="card-image"
    data-testid="code2"
    src="image2"
  />
</div>
`;
